### ISSUE TITLE
Service links are for the wrong device

### ISSUE THREAD

**Issue by jasonehines** [jokob-sk/Pi.Alert#496]
## Describe the issue
When I go to a devices plugin page and select "Services & Ports"  the ports are for a different device. For example when I select the device name "SurgeX", which has a static IP of 10.0.0.22, the ports shown and the links for them are actually my firewall at 10.0.0.1.

> When submitting an issue ❗[enable debug](https://github.com/jokob-sk/Pi.Alert/blob/main/docs/DEBUG_TIPS.md)❗ and [have a look at the docs](https://github.com/jokob-sk/Pi.Alert/tree/main/docs)

there is nothing in the error log

## Paste your `pialert.conf` (remove personal info)

attached as file because GitHub complained it was too long
[pialert_conf.txt](https://github.com/jokob-sk/Pi.Alert/files/13301091/pialert_conf.txt)

## Paste your `docker-compose.yml` and `.env` (remove personal info)

`docker-compose.yml` 

```
name: pi-alert
services:
  app:
    cpu_shares: 90
    command: []
    deploy:
      resources:
        limits:
          memory: 7801M
    environment:
      - HOST_USER_GID=1000
      - HOST_USER_ID=1000
      - PGID=1000
      - PUID=1000
      - SCAN_SUBNETS='10.0.0.0/24 --interface=enp2s0'
      - TZ=America/New_York
    image: jokobsk/pi.alert:23
    labels:
      icon: https://raw.githubusercontent.com/pucherot/Pi.Alert/main/docs/img/1_devices.jpg
    ports:
      - target: 20211
        published: "20211"
        protocol: tcp
    restart: always
    volumes:
      - type: bind
        source: /DATA/AppData/pi-alert/config
        target: /home/pi/pialert/config
      - type: bind
        source: /DATA/AppData/pi-alert/db
        target: /home/pi/pialert/db
    x-casaos:
      envs:
        - container: PUID
          description:
            en_us: ""
        - container: PGID
          description:
            en_us: ""
    devices: []
    cap_add: []
    network_mode: host
    privileged: false
    container_name: ""
x-casaos:
  architectures:
    - amd64
    - arm
    - arm64
  author: WisdomSky
  category: Coolstore
  description:
    en_us: WIFI / LAN intruder detector
  developer: ""
  hostname: ""
  icon: https://raw.githubusercontent.com/pucherot/Pi.Alert/main/docs/img/1_devices.jpg
  index: /
  main: app
  port_map: "20211"
  scheme: http
  store_app_id: pi-alert
  tagline:
    en_us: Wifi / Lan Intruder Detector
  thumbnail: https://raw.githubusercontent.com/pucherot/Pi.Alert/main/docs/img/1_devices.jpg
  title:
    custom: ""
    en_us: Pi.alert

```

`.env` 

```
not using one
```

## Screenshots
<img width="1440" alt="Screenshot 2023-11-08 at 3 18 50 PM" src="https://github.com/jokob-sk/Pi.Alert/assets/870809/b7a1ec20-577f-4fde-b8ee-080b83b91af6">
<img width="1440" alt="Screenshot 2023-11-08 at 3 18 43 PM" src="https://github.com/jokob-sk/Pi.Alert/assets/870809/831f8ee5-a23b-4a3b-b2e4-a0359751e294">

## log
[pialert.log](https://github.com/jokob-sk/Pi.Alert/files/13301095/pialert.log)

**Comment by jasonehines** [comment]
I opened up a copy of the db and it definitely has the wrong MAC address for the entry. Here's an example:
```
Index | Plugin | Object_PrimaryID | Object_SecondaryID | DateTimeCreated | DateTimeChanged | Watched_Value1 | Watched_Value2 | Watched_Value3 | Watched_Value4 | Status | Extra | UserData | ForeignKey
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
11567 | NMAP | 00:0d:ad:02:10:7f | 22/tcp | 2023-11-08 02:08:50 | 2023-11-08 02:08:50 | open | ssh | 10.0.0.1:22/tcp |   | new |   | null | 00:0d:ad:02:10:7f


```

**Comment by jasonehines** [comment]
I think the sql query change will resolve this, please double check me.

**Comment by jokob-sk** [comment]
Thanks for the report and PR - appreciated - will have a look at my end

**Comment by jokob-sk** [comment]
Released in 23.11.11
